rotkehlchen/chain/ethereum/modules/curve/decoder.py

```diff
@@ -258,8 +262,40 @@ def _decode_curve_events(
 
         return None, []
 
-    @staticmethod
+    def _decode_curve_gauge_events(
+            self,
+            tx_log: EvmTxReceiptLog,
+            transaction: EvmTransaction,  # pylint: disable=unused-argument
+            decoded_events: list['EvmEvent'],
+            all_logs: list[EvmTxReceiptLog],  # pylint: disable=unused-argument
+            action_items: Optional[list[ActionItem]],  # pylint: disable=unused-argument
+    ) -> tuple[Optional['EvmEvent'], list[ActionItem]]:
+        if tx_log.topics[0] in (GAUGE_DEPOSIT, GAUGE_WITHDRAW):
+            provider = hex_or_bytes_to_address(tx_log.topics[1])
+            gauge_address = tx_log.address
+            raw_amount = hex_or_bytes_to_int(tx_log.data)
+            for event in decoded_events:
+                crypto_asset = event.asset.resolve_to_crypto_asset()
+                if (
+                    event.location_label == provider and
+                    event.address == gauge_address and
+                    event.balance.amount == asset_normalized_value(amount=raw_amount, asset=crypto_asset)  # noqa: E501
+                ):
+                    if tx_log.topics[0] == GAUGE_DEPOSIT:
+                        event.event_type = HistoryEventType.DEPOSIT
+                        event.event_subtype = HistoryEventSubType.DEPOSIT_ASSET
+                        event.notes = f'Deposit {event.balance.amount} {crypto_asset.symbol} into curve gauge'  # noqa: E501
```

Comment:
Add more information to know which gauge name or address it is. Name is better if you can find it, otherwise address

```suggestion
                        event.notes = f'Deposit {event.balance.amount} {crypto_asset.symbol} into {gauge_name_or_address} curve gauge'  # noqa: E501
```

Reply:
Hmm no, probably can't know gauge name without querying the chain

Reply:
Can't it be saved as part of the cache?

Reply:
Technically is of course possible, but we will probably need either:
- A reverse mapping `gauge address` -> `pool address` and use pools coins symbols as the gauge name.
- Or a reverse mapping `gauge address` -> `pool address` -> `lp token` and use lp token name/symbol.

Of course it is doable, but is it worth it?

Reply:
Ah so there is no name for the gauge? Can't you just have a pool name then instead? Since this is needed anyway? And then by mapping pool to gauge you also have the pool name.

And is it worth it? Doesn't sound hard to me. And the gain is big yes. If the user has to click to etherscan to better understand what's happening we have lost.

Reply:
Alright. Pools also don't have names btw :), but I can make names generated from symbols of pool tokens. Does it sound good to you?

Reply:
For this PR let's just add the address of the gauge. For pools you also use the address I think, right? And then maybe let's make an issue/note for this, since it should not be limited to curve.

Reply:
> For pools you also use the address I think, right?

Yes.

Made an issue: #5809